addons/api/addon/handlers/client-daemon-handler.js

```diff
@@ -75,14 +76,48 @@ const ClientDaemonHandler = {
           'query',
         );
 
-        // TODO: Add some pagination
+        // Return the raw data if we don't push to the store.
+        let records = normalizedPayload.data;
+        if (pushToStore) {
+          records = store.push(normalizedPayload);
+        }
 
-        return store.push(normalizedPayload);
+        // Use a proxy array to store the metadata so the store's `query`
+        // array requirement is satisfied.
+        return ArrayProxy.create({
+          content: records,
+          meta: { totalItems: results?.length ?? 0 },
+        });
       }
       default:
         return next(context.request);
     }
   },
 };
 
+/**
+ * Takes an array and the current page and pagesize to calculate the correct
+ * number of results to return to the caller.
+ *
+ * @param array
+ * @param page
+ * @param pageSize
+ * @returns {[*]}
+ */
+const paginateResults = (array, page, pageSize) => {
+  const length = array?.length;
+  if (!array || length === 0) {
```

Comment:
do we need to check both array and array length here? can this be just `if (length === 0) { return []}`

Reply:
Yeah, array could be undefined/null which I want to catch